src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx

```diff
@@ -29,102 +30,143 @@ const UserMenu: FC<UserMenuProps> = ({
   const { formatMessage } = useIntl();
   const { connectWallet } = useAppContext();
   const { name, profile } = user || {};
+  const [activeSubmenu, setActiveSubmenu] = useState<string | null>(null);
 
   const iconSize = isMobile ? 'small' : 'extraTiny';
 
   return (
     <PopoverBase
       setTooltipRef={setTooltipRef}
       tooltipProps={tooltipProps}
-      classNames={clsx('p-6 rounded-lg flex border-gray-100 justify-start z-[9999] tooltip-container', {
-        'w-full border-none shadow-none': isMobile,
-        'w-[20.125rem]': !isMobile,
-        'h-[32rem] md:h-[23rem]': !isWalletConnected,
-        'md:h-[29rem]': isWalletConnected,
-      })}
-    >
-      {isWalletConnected ? (
-        <WalletConnectedTopMenu
-          userName={profile?.displayName || name}
-          isVerified={isVerified}
-          walletAddress={walletAddress}
-          userReputation={userReputation}
-          totalReputation={totalReputation}
-          nativeToken={nativeToken}
-          avatar={profile?.thumbnail || profile?.avatar || ''}
-          user={user}
-        />
-      ) : (
-        <>
-          <div className={styles.mobileButtons}>
-            <Button mode="tertiaryOutline" isFullRounded>
-              <Icon name="cardholder" appearance={{ size: 'tiny' }} />
-              <p className="text-sm font-inter font-medium ml-1">{formatMessage({ id: 'connectWallet' })}</p>
-            </Button>
-            <Button mode="tertiaryOutline" isFullRounded>
-              <Icon name="list" appearance={{ size: 'extraTiny' }} />
-              <p className="text-sm font-inter font-medium ml-1">{formatMessage({ id: 'help' })}</p>
-            </Button>
-          </div>
-          <div className="w-full pb-6 mb-6 border-b border-b-gray-200 md:pb-5 md:mb-5">
-            <Button mode="quaternaryOutline" isFullSize onClick={connectWallet}>
-              {formatMessage({ id: 'connectWallet' })}
-            </Button>
-          </div>
-        </>
+      classNames={clsx(
+        'p-6 rounded-lg flex border-gray-100 justify-start z-[9999] tooltip-container relative overflow-hidden',
+        {
+          'w-full border-none shadow-none': isMobile,
+          'w-[20.125rem]': !isMobile,
+          'h-[32rem] md:h-[23rem]': !isWalletConnected && !activeSubmenu,
```

Comment:
This style is used twice

Reply:
One is for connected and second for not connected wallet -> `!isWalletConnected && !activeSubmenu` and `isWalletConnected && !activeSubmenu`